packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -797,14 +771,44 @@ const VirtualListBaseFactory = (type) => {
 			delete rest.spacing;
 			delete rest.updateStatesAndBounds;
 
+			const
+				{dataSize, direction, itemSize, overhang, rtl, spacing} = this.prevProps,
+				{props} = this,
+				hasMetricsChanged = (
+					direction !== props.direction ||
+					((itemSize instanceof Object) ? (itemSize.minWidth !== props.itemSize.minWidth || itemSize.minHeight !== props.itemSize.minHeight) : itemSize !== props.itemSize) ||
+					overhang !== props.overhang ||
+					spacing !== props.spacing
+				);
+
+			this.hasDataSizeChanged = (dataSize !== props.dataSize);
+
+			if (hasMetricsChanged || this.props.clientSize && this.primary === null && this.secondary === null) {
+				this.calculateMetrics(props);
+				this.updateStatesAndBounds(props, true);
+				this.setContainerSize();
+			} else if (this.hasDataSizeChanged) {
+				this.updateStatesAndBounds(props, true);
+				this.setContainerSize();
+			} else if (rtl !== props.rtl) {
+				const {x, y} = this.getXY(this.scrollPosition, 0);
+
+				this.cc = [];
+				if (type === Native) {
+					this.scrollToPosition(x, y, props.rtl);
```

Comment:
If `rtl` mode changes, we need to update `VirtualList` as well as `VirtualList items` based on the `rtl` mode.

For `VirtualList`, we called the `this.scrollToPosition` or `this.setScrollPosition` here. There might be no problem if calling it in `componentDidUpdate`.

But for `VirtualList items`, `this.cc = [];` was executed at https://github.com/enactjs/enact/blob/31e372d4d38972d7e0448a0f4481a6502dde2efe/packages/ui/VirtualList/VirtualListBase.js#L796, then `this.positionItems();` was called at https://github.com/enactjs/enact/blob/31e372d4d38972d7e0448a0f4481a6502dde2efe/packages/ui/VirtualList/VirtualListBase.js#L805 If moving the `this.cc =[];` to `componentDidUpdate`,
- To render the items, we need to call `this.forceUpdate()` instead of `this.setState()` in `componentDidUpdate` because no state changes. I think calling `forceUpdate` is not a good pattern.
- `render()` function will be called once more.

So I thought it would be better to execute `this.cc = [];` in `render`() with calling `scrollTo` together.

Reply:
Is `this.cc` required to be reset when `rtl` prop changes? Doesn't index and key stay the same in RTL layout? Or am I missing something?

Reply:
All items usually display properly depending on the RTL mode based on CSS. But app developers could position the content in the item depending on the RTL mode. For this case, we need to re-render the items so that app developers could position them properly in `itemRender`.

Reply:
But there is no real case until now. Let's move this logic to `componentDidUpdate`.

Reply:
<On Feb 18th>
- The code related with RTL including calling `scrollTo` is moved to not render phase but `componentDidUpdate`.
- The code `this.cc = [];` that we discussed is removed. If any issue raised, then we could revisit it.